### ISSUE TITLE
No way to add custom permissions (non add/change/delete/view) to the group edit form

### ISSUE THREAD

**Issue by biki** [wagtail/wagtail#8086]
### Issue Summary

I added two custom views with menu items to the Wagtail backend. I now wanted to secure these views with custom permissions. There is no underlying django.Model behind these views which make the permission handling more difficult.
My end goal was to display these permissions under the "Other permissions" in the group edit forms just like "Can access Wagtail admin".

### Steps to Reproduce

1. Created two custom permissions with a migration. I put them under the `wagtailadmin` app_label and `admin` model.
```
content_type = ContentType.objects.get(app_label="wagtailadmin", model="admin")
Permission.objects.create(
    content_type=content_type, codename="branch_import", name="Branch-Import"
)
Permission.objects.create(
    content_type=content_type, codename="hansalog_import", name="Hansalog-Import"
)
```
2. Added the hook like the documentation said
```
@hooks.register("register_permissions")
def register_permissions():
    app = "wagtailadmin"

    return Permission.objects.filter(
        content_type__app_label=app,
        codename__in=["hansalog_import", "branch_import"],
    )
```
3. The permissions do not show up.
4. In the end, I had to overwrite / use a custom version of `format_permissions` found here: https://github.com/wagtail/wagtail/blob/716bf92c2dc2da2aca5e8f5aa6768b5b087cd4b0/wagtail/users/templatetags/wagtailusers_tags.py because there seems to be no way to add additional entries to the "Other permissions" section otherwise, as line 60 (https://github.com/wagtail/wagtail/blob/716bf92c2dc2da2aca5e8f5aa6768b5b087cd4b0/wagtail/users/templatetags/wagtailusers_tags.py#L60) is directly looking at the first element only.

When I slighty change the code to the following, the new permissions show up, although with a new empty row at the very top of the page.
```
for perm in content_perms:
      if perm.content_type.name == "admin":
          other_perms.append((perm, checkboxes_by_id[perm.id]))
          continue
```

Is there something I am missing or doing wrong? Is it not intended to be able to add our own permissions to the "Other permissions" are in the group add/edit forms?

---

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: no

### Technical details

* Python version: 3.8.1
* Django version: 3.2.12
* Wagtail version: 2.16.1
* Browser version: Brave / Chromium engine

**Comment by urlsangel** [comment]
Also having the very same issue when trying to make Wagtail 2FA optional, as per the docs here:

https://wagtail-2fa.readthedocs.io/en/latest/#making-2fa-optional

This [block of code](https://github.com/wagtail/wagtail/blob/c111deef03c02c24941e06d44274a6ddf83556e4/wagtail/users/templatetags/wagtailusers_tags.py#L60):

```
      if content_perms[0].content_type.name == "admin":
            perm = content_perms[0]
            other_perms.append((perm, checkboxes_by_id[perm.id]))
            continue
```

The continue statement here means that only the first item is ever added, so the permission `<Permission: wagtailadmin | admin | Enable 2FA>` is never added to the form.

**Comment by laymonage** [comment]
Removing the modeladmin label as I think this is not related to ModelAdmin.

**Comment by lb-** [comment]
A PR was opened that made some progress here, it would need unit tests and needs to pass all CI checks.

Anyone is free to pick up where this PR left off and see if they are able to get this resolved.

https://github.com/wagtail/wagtail/pull/10778

**Comment by laymonage** [comment]
Thank you for the report (and sorry it took so long for us to get back to this)!

I can confirm that this behaviour exists. I did notice the bug in the code a while back while working on #9711, but I forgot to revisit this. I think your solution is correct here, we should iterate all the permissions for Wagtail's `admin` content type, not just taking the first one.

I'm not sure if this should be categorised as a bug or an enhancement, though. But I agree that this is something that would be nice to have.

**Comment by gasman** [comment]
Fixed in #11668
